collection-scripts/gather

```diff
@@ -4,48 +4,74 @@ unset KUBECONFIG
     fi
   done
+  echo "Will collect debug info from migclusters [${clusters[@]}]"
 
+  # Iterate over all connected non-host OpenShift clusters
   for cluster in ${clusters[@]}; do
     unset KUBECONFIG
     if [ ${cluster} != "host" ]; then
+      echo "[cluster=${cluster}] Performing token login"
       token=$(oc get -n $(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.serviceAccountSecretRef.namespace }}') secret $(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.serviceAccountSecretRef.name }}') -o go-template='{{ .data.saToken }}' | base64 -d)
       url=$(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.url }}')
       export KUBECONFIG=/tmp/kubeconfig
       oc login --insecure-skip-tls-verify=true --token=${token} $url
       if [ "$?" != "0" ]; then
         object_collection_path="/must-gather/${cluster}/"
         mkdir -p ${object_collection_path}
-        echo "login to cluster ${cluster} unsuccessful."  > "${object_collection_path}/login_error.txt"
+        echo "Login to migcluster ${cluster} unsuccessful."  > "${object_collection_path}/login_error.txt"
         continue
       fi
     fi
 
-    # Migration namespace list
+    # Build list of namespaces where MTC is installed
     namespaces=()
-
     for ns in $(/usr/bin/oc get migrationcontrollers.migration.openshift.io --all-namespaces --no-headers | awk '{print $1}')
     do
-      namespaces+=(ns/$ns)
+      echo "[cluster=${cluster}][namespace=${ns}] Detected MTC installation"
+      namespaces+=(${ns})
     done
 
-    # Run the collection of resources using must-gather
+    # Collect resources in MTC namespaces with must-gather
     for ns in ${namespaces[@]}; do
-      /usr/bin/oc adm inspect --dest-dir must-gather/${cluster} --all-namespaces ${ns}
+      echo "[cluster=${cluster}][namespace=${ns}] Running oc adm inspect"
+      /usr/bin/oc adm inspect --dest-dir must-gather/${cluster} --all-namespaces ns/${ns} &
     done
 
-  done
-  # Collect the migration and velero CRs
-  /usr/bin/gather_crs
+    # Collect the migration and velero CRs
+    echo "[cluster=${cluster}] Gathering MTC and Velero CRs for namespaces [${namespaces[@]}]"
+    /usr/bin/gather_crs ${cluster} ${namespaces} &
 
-  # Collect the logs
-  /usr/bin/gather_logs
+    # Collect the logs"
+    echo "[cluster=${cluster}] Gathering logs for namespaces [${namespaces[@]}]"
+    /usr/bin/gather_logs ${cluster} ${namespaces} &
+  done
 
   # Collect metrics from Prometheus
-  /usr/bin/gather_metrics
+  echo "[cluster=host] Gathering prometheus metrics"
+  /usr/bin/gather_metrics &
+
+  # Waits for gather_crs, gather_logs, gather_metrics running in background
+  echo "Waiting for background gather tasks to finish"
+  wait
 done
+
+# Tar all must-gather artifacts for faster transmission 
+echo "Tarring must-gather artifacts..."
+archive_path="/must-gather-archive"
+mkdir -p ${archive_path}
+tar -zcf ${archive_path}/must-gather.tar.gz /must-gather/
```

Comment:
Tarring before transfer over the network has a big benefit (>10x filesize reduction)